hilti/toolchain/src/base/util.cc

```diff
@@ -6,17 +6,27 @@
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <unistd.h>
+#include <utf8proc/utf8proc.h>
 
 #include <algorithm>
 #include <cstdlib>
 #include <cstring>
 
 #include <hilti/rt/backtrace.h>
 
-#include <hilti/3rdparty/pathfind/PathFind.hpp>
-#include <hilti/3rdparty/utf8proc/utf8proc.h>
 #include <hilti/base/logger.h>
 
+// There's a bizarr "printf" inside the PathFind code. So we include the
+// source file here to be able to #define that away.
+#define __ignore__(x,y)
+#define printf __ignore__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
+#include <pathfind/src/PathFind.cpp>
+#pragma GCC diagnostic pop
+#undef printf
+#undef __ignore__
```

Comment:
Let's instead introduce a `pathfind.hpp` header which does this magic and then use that one exclusively.

Reply:
We use this only at this one place, it's really just an implementation detail.